src/main/java/com/google/api/generator/gapic/composer/SettingsSampleCodeComposer.java

```diff
@@ -0,0 +1,162 @@
+                TypeNode.withReference(ConcreteReference.withClazz(Duration.class)))
+            .setMethodName("ofSeconds")
+            .setArguments(
+                ValueExpr.withValue(
+                    PrimitiveValue.builder().setType(TypeNode.INT).setValue("30").build()))
+            .build();
+
+    MethodInvocationExpr timeoutBuilderMethodExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(
+                MethodInvocationExpr.builder()
+                    .setExprReferenceExpr(
+                        MethodInvocationExpr.builder()
+                            .setExprReferenceExpr(retrySettingsMethodExpr)
+                            .setMethodName("getRetrySettings")
+                            .build())
+                    .setMethodName("toBuilder")
+                    .build())
+            .setMethodName("setTotalTimeout")
+            .setArguments(Arrays.asList(timeoutArExpr))
+            .build();
+
+    MethodInvocationExpr retrySettingsArgExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(timeoutBuilderMethodExpr)
+            .setMethodName("build")
+            .build();
+
+    MethodInvocationExpr settingBuilderMethodExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(retrySettingsMethodExpr)
+            .setMethodName("setRetrySettings")
+            .setArguments(Arrays.asList(retrySettingsArgExpr))
+            .build();
+
+    VariableExpr settingsVarExpr =
+        VariableExpr.withVariable(
+            Variable.builder()
+                .setType(classType)
+                .setName(getServiceSettingsName(className))
+                .build());
+
+    AssignmentExpr settingBuildAssignmentExpr =
+        AssignmentExpr.builder()
+            .setVariableExpr(settingsVarExpr.toBuilder().setIsDecl(true).build())
+            .setValueExpr(
+                MethodInvocationExpr.builder()
+                    .setExprReferenceExpr(localSettingsVarExpr)
+                    .setMethodName("build")
+                    .setReturnType(classType)
+                    .build())
+            .build();
+
+    List<Statement> statements =
+        Arrays.asList(initLocalSettingsExpr, settingBuilderMethodExpr, settingBuildAssignmentExpr)
+            .stream()
+            .map(e -> ExprStatement.withExpr(e))
+            .collect(Collectors.toList());
+    return SampleCodeJavaFormatter.format(writeStatements(statements));
+  }
+
+  private static String getServiceSettingsName(String className) {
+    return JavaStyle.toLowerCamelCase(className).replace(STUB, EMPTY_STRING);
+  }
+
+  private static String getClassSettingsBuilderName(String className) {
+    return JavaStyle.toLowerCamelCase(
+            String.format(BUILDER_NAME_PATTERN, JavaStyle.toLowerCamelCase(className)))
+        .replace(STUB, EMPTY_STRING);
+  }
+
+  private static String writeStatements(List<Statement> statements) {
```

Comment:
Same as in #493, if we centralize this in one class, we can avoid redefining a method. Also, for #493, this can be an overloaded method.

Reply:
Thanks for pointing out. Add a method with its unit test in #499